code/object/objcollide.cpp

```diff
@@ -1021,466 +625,430 @@ void obj_reset_colliders()
-				return;
-			}
-		}
-	} else {
-		//if ( A->type == OBJ_BEAM ) {
-			//if(beam_collide_early_out(A, B)){
-				//collision_info->next_check_time = -1;
-				//return;
-			//}
-		//}
-
-		// only check debris:weapon collisions for player
-		if (check_collision == collide_debris_weapon) {
-			// weapon is B
-			if ( !(Weapon_info[Weapons[B->instance].weapon_info_index].wi_flags[Weapon::Info_Flags::Turns]) ) {
-				// check for dumbfire weapon
-				// check if debris is behind laser
-				float vdot;
-				if (Weapon_info[Weapons[B->instance].weapon_info_index].subtype == WP_LASER) {
-					vec3d velocity_rel_weapon;
-					vm_vec_sub(&velocity_rel_weapon, &B->phys_info.vel, &A->phys_info.vel);
-					vdot = -vm_vec_dot(&velocity_rel_weapon, &B->orient.vec.fvec);
-				} else {
-					vdot = vm_vec_dot( &A->phys_info.vel, &B->phys_info.vel);
-				}
-				if ( vdot <= 0.0f )	{
-					// They're heading in opposite directions...
-					// check their positions
-					vec3d weapon2other;
-					vm_vec_sub( &weapon2other, &A->pos, &B->pos );
-					float pdot = vm_vec_dot( &B->orient.vec.fvec, &weapon2other );
-					if ( pdot <= -A->radius )	{
-						// The other object is behind the weapon by more than
-						// its radius, so it will never hit...
-						collision_info->next_check_time = -1;
-						return;
-					}
-				}
-
-				// check dist vs. dist moved during weapon lifetime
-				vec3d delta_v;
-				vm_vec_sub(&delta_v, &B->phys_info.vel, &A->phys_info.vel);
-				if (vm_vec_dist_squared(&A->pos, &B->pos) > (vm_vec_mag_squared(&delta_v)*Weapons[B->instance].lifeleft*Weapons[B->instance].lifeleft)) {
-					collision_info->next_check_time = -1;
-					return;
-				}
-
-				// for nonplayer ships, only create collision pair if close enough
-				if ( (B->parent >= 0) && !((Objects[B->parent].signature == B->parent_sig) && (Objects[B->parent].flags[Object::Object_Flags::Player_ship])) && (vm_vec_dist(&B->pos, &A->pos) < (4.0f*A->radius + 200.0f)) ) {
-					collision_info->next_check_time = -1;
-					return;
-				}
-			}
-		}
-
-		// don't check same team laser:ship collisions on small ships if not player
-		if (check_collision == collide_ship_weapon) {
-			// weapon is B
-			if ( (B->parent >= 0)
-				&& (Objects[B->parent].signature == B->parent_sig)
-				&& !(Objects[B->parent].flags[Object::Object_Flags::Player_ship])
-				&& (Ships[Objects[B->parent].instance].team == Ships[A->instance].team) 
-				&& (Ship_info[Ships[A->instance].ship_info_index].is_small_ship()) 
-				&& (Weapon_info[Weapons[B->instance].weapon_info_index].subtype == WP_LASER) ) {
-				collision_info->next_check_time = -1;
-				return;
-			}
-		}
+	sort_list_z.clear();
+	{
+		TRACE_SCOPE(tracing::SortColliders);
+		obj_quicksort_colliders(&sort_list_y, 0, (int)(sort_list_y.size() - 1), 1);
 	}
+	obj_find_overlap_colliders(sort_list_z, sort_list_y, 1, false);
 
-	obj_pair new_pair;	
-
-	new_pair.a = A;
-	new_pair.b = B;
-	new_pair.check_collision = check_collision;
```

Comment:
Could you please help me understand why it's OK to remove this assignment?

Reply:
They have been moved to lines 963ff.
When I extracted local functions to the anonymous namespace, I moved them before their point of use.

The github diff viewer fails us here.

Reply:
I can see these three lines there
```
    new_pair.a = A;
    new_pair.b = B;
    new_pair.next_check_time = collision_info->next_check_time;
```
but I can't see 
```
	new_pair.check_collision = check_collision;
```

(btw - I noticed this when using the github diff viewer that ignores whitespace)

Reply:
Yeah, sorry that is confusing (I had to backtrack as well)
That field was unused and has been removed in this commit as well.

see objcollide.h line 57

I will add that to the commit message.